src/common/time/src/timestamp.rs

```diff
@@ -143,7 +143,8 @@ impl Timestamp {
             Some(Timestamp::new(value, unit))
         } else {
             let mul = unit.factor() / self.unit().factor();
-            Some(Timestamp::new(self.value.div_ceil(mul as i64), unit))
+            let new_ts = self.value as f64 / mul as f64;
```

Comment:
I think we should port the implementation from https://github.com/rust-lang/rust/pull/88582

Reply:
Thanks for searching for this.